### PR TITLE
optionally cache failed mvi profile responses with separate ttl in redis

### PR DESCRIPTION
**Pull request by dhall**
## Description of change

Our error counts are inflated because we don't cache a failed response when building a user's profile from MVI. This PR retains the 24h cache for successful profile lookups and introduced a 30 minute cache for failed lookups. I believe this will largely reduce the error count since the profile is referenced frequently in the app. As an example the process of signing in and rendering the landing page makes 7 mvi requests when there is a failed lookup scenario compared to 1 for a success (due to caching that good response). I introduced the different ttl for failures so that in the event the profile becomes available we aren't locking out a user from getting the newest information. 

## Testing done

Local testing by logging in with different users, forcing different failure scenarios by raising various exceptions, recording requests/responses to mvi and checking redis for cache entries and ttl's of those cache entries. Also rspec test coverage.

## Testing planned

These changes will be turned off by default so I'll create a PR to turn this on in staging for end to end testing.

## Acceptance Criteria (Definition of Done)
When the toggle is enable, both success and failures are cached to redis.

#### Unique to this PR

- [x] Allow caching of failed mvi requests via feature flag

#### Applies to all PRs

- [x] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Comment by saneshark** [comment]
@dhall was it not trivial to use the same redis cache to store failures, but to introduce a new schema to how mvi treats it as "cached" vs "uncached?" In other words, just store the failures in the same key in redis, but have the object that is stored be identified as something that will be requested multiple times (no 30 minute throttle), and then using tags you can determine if it is the first time it encountered a failure or the second time. 

I don't want our existing dashboard to change signficantly. I would prefer if we could introduce a new dashboard side by side that incorporates the tags which would eventually replace what we have today.

But if there was challenges to that approach, lets discuss.

**Comment by dhall** [comment]
The approach of this PR is less involved than the other approach of caching but ignoring that cache from a service perspective. It would involve some changes to the caching mechanism (cache_aside). I actually like the idea of not hammering mvi for the failures.. even if we drop the ttl to 10 or even 5 minutes we would see benefits from a noise perspective. I was thinking the original intent of preserving the dashboard was so we could verify we were making a positive impact on the problem without making too many changes at once. Now that we've got good evidence the orchestration is making a difference I feel like we don't need to preserve the original dashboard's error noise.

File: app/models/mvi.rb [hunk 1]
```diff
@@ -120,11 +121,23 @@ def mvi_response
 
   def response_from_redis_or_service
     do_cached_with(key: @user.uuid) do
-      mvi_service.find_profile(@user)
+      profile = mvi_service.find_profile(@user)
+      self.class.redis_namespace_ttl = set_ttl(profile)
+      profile
```
Comment (omgitsbillryan) [inline app/models/mvi.rb]: `redis_namespace_ttl` is a class var that's being modified through an instance method, so I think it's a thread safety issue. Granted, I guess the worst thing that could happen would be a failed response saved for `each_ttl` or a successful response stored for `failure_ttl`. Not all that dangerous for right now, but just an observation.

Comment (dhall) [reply app/models/mvi.rb]: Thanks for pointing this out @omgitsbillryan I'll keep an eye on things and see if it becomes a problem.

**Pull request merged**